File: cmd/connectGateway.ts
import getDeviceList from "./deviceList";
import { exec, ip_pattern } from "./utils";

/**
 * Scans the ip configuration of the computer and
 * @returns 
 */
export default function connectGateway(): Device[] {
    const data = exec({
        win: "ipconfig",
        default: "ifconfig"
    })
    if (data.error) throw new Error(data.stderr)
    // extract the ip address from the list of gateways
    const ips = data.stdout.match(new RegExp(ip_pattern, "g"))
    // connect to each
    if (ips) ips.forEach((ip) => exec(`adb connect ${ip}`))
    return getDeviceList();
}

File: go/cli/connect_gateway.go
package cli

import (
	"fmt"
	"regexp"
	"roidy/go/utils"
)

// scan the local network and try connecting to all gateways, assuming one is an available android devices
func ConnectGateway() ([]Device, error) {
	win := "ipconfig"
	data, err := utils.Exec(utils.Cmd{
		Win: &win,
		Cmd: "ifconfig",
	})
	if err != nil {
		return nil, err
	}
	// get ip address
	ipAddrs := regexp.MustCompile(`\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}`).FindAllString(data, -1)
	for _, ip := range ipAddrs {
		utils.Execs(fmt.Sprintf("adb connect %s", ip))
	}
	devices, err := ListDevices()
	return devices, err
}
